BREAKING.md

```diff
@@ -19,10 +19,16 @@ It's important to communicate breaking changes to our stakeholders. To write a g
 
 ## 2.0.0-internal.3.0.0 Upcoming changes
 - [Deprecated IPendingFlush](#Deprecated-IPendingFlush)
+- [Removing internal connection details from `IConnectionDetails`](#Removing-internal-connection-details-from-IConnectionDetails)
 
 ### Deprecated IPendingFlush
 `IPendingFlush` has been deprecated. Use batch metadata on `IPendingMessage` instead to indicate the end of a batch.
 
+### Removing internal connection details from `IConnectionDetails`
+
+Removing `mode`, `version` and `initialClients` from `IConnectionDetails`, no longer exposing these to runtime. They will only be used internally. Removing `existing` altogether as it no longer provides value.
```

Comment:
I'm not sure about this note. It doesn't say why they're removed and also doesn't mention a replacement or a workaround (reading it from a customer's POV). If there's no workaround, then it might be that the properties should not be used in the first place?

> Removing `existing` altogether as it no longer provides value.

Doesn't say much about it..

Reply:
> a replacement or a workaround

+1

The current guidance is that before anything is removed from a public API, we need to mark it as deprecated and release that deprecation in a minor release, including an "Upcoming" note with guidance for what to do instead.  This way when a partner takes that minor bump, they have everything they need to move off the deprecated API, and by the time they are doing the major bump it's a no-op

Reply:
In this case, it's possible we already missed the window to remove anything else in internal.3.0.  But maybe not, I will get clarity with the team and circle back.

In any case, starting with the change in `main` that announces the deprecation and provides guidance on a workaround can happen any time.  You just may need to wait on this `next` PR until after this major release.

Reply:
yeah the properties should not be used in the first place by customers, they are moved to a new internal interface but i didn't want to mention that because customers don't need to know about internal stuff?

Reply:
Yes, we should not mention that they are available internally. We should mention they are gone, and the reason why they are gone / any replacement. Here is my short summary:
- exisitng: always true, no longer carries any information.
- mode: this is implementation detail of connection.
- initialClients, version: implementation detail / part of handshake protocol of establishing connection. This data should not be accessble.